lang-guide/chapters/pipelines.md

```diff
@@ -81,3 +81,149 @@ nu demo.nu o> file.txt o+e>| str upcase
 ```
 
 Also note that `complete` is special, it doesn't work with `e>|`, `o+e>|`.
+
+## Stdio and redirection behavior examples
+
+Pipeline and redirection behavior can be hard to follow when they are used with subexpression, or custom commands.  Here are some examples to show stdio behavior
+
+### Examples for subexpression
+
+- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4)
+
+| Command | Stdout     | Stderr     |
+| ------- | --------   | ---------- |
+| cmd1    | Piped      | Terminal   |
+| cmd2    | *Terminal* | Terminal   |
+| cmd3    | Piped      | Terminal   |
+| cmd4    | Terminal   | Terminal   |
+
+- (^cmd1 | ^cmd2; ^cmd3 | ^cmd4) | ^cmd5
```

Comment:
I wonder if it might be good to show where the output of cmd5 goes as well but I'm not sure you can tell that without understanding what cmd5 does. e.g. does it write to file, stdout, or stderr? 

Maybe if each table had a short description saying what it demonstrates then the user would know that cmd5 writes to stdout (or whatever the example shows) and it wouldn't be needed in the table?

Reply:
> Maybe if each table had a short description saying what it demonstrates

I'd like to suggest leaving it there, the example command themselves only care about the behavior of cmd(1/2/3/4).  The behavior and output of cmd5 is not important.

Reply:
I think it would be better to have a short description above each table for clarity, but we can start with what we have here if you like.

Reply:
Sure.  I updated a short description for these commands